ppx-lib/gen_js_api_ppx.ml

```diff
@@ -1284,8 +1287,43 @@ and gen_funs ~global_attrs p =
     ) p.ptype_params
   in
   let loc = p.ptype_loc in
-  let typvar_used, of_js, to_js =
+  let typvar_used, of_js, to_js, custom_funs =
     match p.ptype_kind with
+    | _ when has_attribute "js.custom" global_attrs ->
+        let result =
+          match get_attribute "js.custom" global_attrs with
+          | Some (k, PStr structure) ->
+              let assert_function label (vbs: value_binding list) =
+                let check vb =
+                  match vb.pvb_pat.ppat_desc with
+                  | Ppat_var nameloc when nameloc.txt = label -> true
+                  | _ -> false
+                in
+                if List.exists check vbs then ()
+                else raise (Error (k.loc, Missing_requried_definitions label))
+              in
+              let rec get_value_bindings (s: structure) =
+                List.concat_map (fun (si: structure_item) ->
+                  match si.pstr_desc with
+                  | Pstr_value (_, vbs) -> vbs
+                  | Pstr_include incdec ->
+                    let result =
+                      match incdec.pincl_mod.pmod_desc with
+                      | Pmod_structure s' -> get_value_bindings s'
+                      | _ -> []
+                    in result
+                  | _ -> []
+                ) s
+              in
+              let vbs = get_value_bindings structure in
```

Comment:
It feels a bit weird to inject any value declaration (toplevel or under an included literal struct -- is that case useful?) into an implicitly recursive binding, and ignore (without complaining) all others structure items.

Another direction would be to use a more structured payload, e.g a record with two fields:

````ocaml
[@@js.custom { of_js = (fun ... -> ...); to_js = (fun ... -> ...) }]
````
(in your example:

````ocaml
[@@js.custom { of_js = Obj.magic; to_js = Obj.magic }]
````

I'm not opposed to keeping the current approach (with two value bindings), but then I'd prefer failing explicitly on any other structure item.  Or do see cases where allowing extra value bindings  can be useful?

Reply:
I used a structure here because we can get the `value_binding`s directly from the `PStr`.  If we used a record here, we have to manually convert `Pexp_fun` to `value_binding`, which can be a little pain. I think it's easier to just reject any structure items other than `_to_js` and `_of_js`.

Reply:
Implemented with https://github.com/LexiFi/gen_js_api/pull/125/commits/b95f420e92c6912c2a15fa396b7038f816dead83

Reply:
Sorry for the late reply.  I still find it misleading that (1) `let t_of_js = ...  let t_to_js = ...` and (2) `let rec t_of_js = ... and t_to_js = ...` mean the same thing.

If you are concerned with the implementation complexity, I suggest you try with the syntax `{of_js=...; to_js=...}`; it should really not be more complex to implement.

Reply:
Okay, let's try `{of_js=...; to_js=...}` then.

Reply:
Implemented. Please review it 🙏